components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -3321,13 +3321,20 @@ public static String buildCallerPathRedirectURL(String callerPath, Authenticatio
          since My Account is SaaS.
          */
         if (!MY_ACCOUNT_APP.equals(serviceProvider)) {
-            if (callerPath != null && callerPath.startsWith("/t/")) {
+            if (callerPath != null && callerPath.startsWith(FrameworkConstants.TENANT_CONTEXT_PREFIX)) {
                 String callerTenant = callerPath.split("/")[2];
                 String callerPathWithoutTenant = callerPath.replaceFirst("/t/[^/]+/", "/");
                 String redirectURL = ServiceURLBuilder.create().addPath(callerPathWithoutTenant)
                         .setTenant(callerTenant, true)
                         .build().getAbsolutePublicURL();
                 return redirectURL;
+            } else if (callerPath != null && callerPath.startsWith(FrameworkConstants.ORGANIZATION_CONTEXT_PREFIX)) {
+                String callerOrgId = callerPath.split("/")[2];
+                String callerPathWithoutOrgId = callerPath.replaceFirst("/o/[^/]+/", "/");
+                String redirectURL = ServiceURLBuilder.create().addPath(callerPathWithoutOrgId)
+                        .setTenant(callerOrgId, true).setOrganization(callerOrgId)
```

Comment:
Hope we are safe to assume the orgId = tenantDomain when the caller path is organization qualified. Shall we add a comment here.

Reply:
Fixed with 2764a25eb3195ae213c29403c0cf7f912ec0d843. Used the login tenant domain in the context as the tenant domain instead of using the organization id extracted from the caller path.